File: programa.js
import { createInterface } from 'node:readline/promises';
import { stdin as input, stdout as output } from 'node:process';

async function iniciarPrograma() {
    // Configuramos la "entrada" y "salida" de la terminal
    const rl = createInterface({ input, output });
    const personas = [];

    console.log("--- Registro de Alumnos (Escribe 'finalizar' para terminar) ---");

    while (true) {
        const nombre = await rl.question("Nombre: ");
        
        if (nombre.toLowerCase() === 'finalizar') break;

        const edad = parseInt(await rl.question("Edad: "));
        const nota = parseFloat(await rl.question("Nota: "));

        // Guardamos la sublista [nombre, edad, nota] en nuestra matriz
        personas.push([nombre, edad, nota]);
        console.log("Guardado.\n");
    }

    rl.close(); // Cerramos la terminal al terminar

    if (personas.length > 0) {
        // 1. Mostrar lista original
        console.log("\n=== LISTADO COMPLETO ===");
        personas.forEach(p => {
            console.log(`Nombre: ${p[0]} | Edad: ${p[1]} | Nota: ${p[2]}`);
        });

        // 2. Ordenar por nota (índice 2) de mayor a menor
        // Usamos [...personas] para crear una copia y no arruinar la original
        const listaOrdenada = [...personas].sort((a, b) => b[2] - a[2]);

        console.log("\n=== ORDENADO POR NOTA (Mayor a Menor) ===");
        listaOrdenada.forEach(p => {
            console.log(`Nota: ${p[2]} - Estudiante: ${p[0]}`);
        });

        // 3. Promedio
        const suma = personas.reduce((acumulador, p) => acumulador + p[2], 0);
        const promedio = suma / personas.length;
        console.log(`\n>>> PROMEDIO GENERAL: ${promedio.toFixed(2)}`);
    } else {
        console.log("No hay datos.");
    }
}

iniciarPrograma();

File: programa.py
# Lista principal (matriz)
personas = []

while True:
    print("\n--- Nuevo registro ---")
    nombre = input("Ingresa el nombre (o 'finalizar' para terminar): ")
    
    if nombre.lower() == 'finalizar': 
        break
    
    #pide edad y nota
    edad = int(input("Ingresa la edad: "))
    nota = float(input("Ingresa la nota: "))

    # guardamos los datos de ESTA persona
    registro = [nombre, edad, nota]

    # lo metemos en la lista principal
    personas.append(registro) 
    print("Registro agregado correctamente.")

# --- una vez que salimos del bucle (escribiendo finalizar) ---

if len(personas) > 0:
    # Mostrar lista original
    print("\n=== LISTADO COMPLETO (ORDEN DE CARGA) ===")
    for p in personas:
        print(f"Nombre: {p[0]} | Edad: {p[1]} | Nota: {p[2]}")

    # Ordenar de mayor a menor por nota (índice 2)
    lista_ordenada = sorted(personas, key=lambda x: x[2], reverse=True)

    print("\n=== LISTADO ORDENADO POR NOTA (MAYOR A MENOR) ===")
    for p in lista_ordenada:
        print(f"Nota: {p[2]} - Estudiante: {p[0]}")

    # Promedio
    total_notas = sum(p[2] for p in personas)
    promedio = total_notas / len(personas)
    print(f"\n>>> PROMEDIO GENERAL: {promedio:.2f}")
else:
    print("No se cargaron datos.")